### ISSUE TITLE
X Consider not registering new users to season after a certain point [20,13,1.54]

### ISSUE THREAD

**Issue by stenington** [Iridescent-CM/technovation-app#2194]
**Describe the bug**
We have users who seemingly 
1. created their account before the 2019 season, and
2. did not log in at any point in the 2019 season

who are now logging in (presumably interested in the 2020 season) who are getting a "Welcome to Technovation 2019" email describing a season and a set of dates to them that have already passed.

**Expected behavior** (updated 4/22/20)
I think it makes sense to identify a point in the season after which logging in will no longer register you for that current season. With this implementation, that point will apply to all types of users. Any user logging in after that date will not be marked as participating in a TG season, will not receive a welcome email, nor associated resets (e.g. consent form).

**Comment by stenington** [comment]
Related to #2229

**Comment by hellafitz** [comment]
@devincdillon please check with program folks about what date would make sense - and if it'd need to be different for different user groups. Please answer by Monday 10/28 EOD. Thanks!

**Comment by devincdillon** [comment]
Hi Guys,

Sorry I'm just seeing this. Okay, I'm having some problems with understanding this issue.

So, on TG dashboard, students or mentors are logging in but seeing information from an old season? I guess you want me to pick a date that applies to the 2021 season, where if they login they no longer see 2020 information? Or are you wanting me to pick a date for the current season?

If we are talking current season, then I think that date should be like... now. If we are talking future facing... I need to make a 2020-2021 calendar and then project the date, but would have to note that it could change... it's so far out.

**Comment by stenington** [comment]
@devincdillon This issue isn't happening right now, it was happening just before we opened the 2020 season, and it'll start happening again as we approach the end of the 2020 season.

I think of it this way. The 2020 season started Oct. 10, 2019 and let's pretend 2021 starts Oct. 10, 2020. 

- If I log in for the first time on Oct. 11, 2019 then getting an email that welcomes me to the 2020 season makes 100% sense. 
- If I log in for the first time in Nov. or Dec. 2019, or even Jan, Feb 2020, it still makes sense for me to get an email welcoming me to the 2020 season.
- If I log in for the first time on Oct. 9, 2020, it **no longer** makes sense for me to get an email welcoming me to the 2020 season. The 2020 season is over, I'm more than likely coming to the platform interested in the 2021 season.

Basically we send people a nice welcome to the season **all the way until the next season starts**, even when there's no way they could participate in the current season, coming to it so late.

**Comment by devincdillon** [comment]
Thanks Mike that helped a lot.

So I'm unsure if the key dates we show include judging information or not, which would change my answer (I think)

**If it shows judging dates**

- turn off welcome email June 1

**If it does not show judging dates**

- turn off April 20 (which is when submission closes)

**Comment by stenington** [comment]
It seems like the idea that you're always in a season is deeply integrated into the app. For example, our Account and TeamSubmission factories rely on `RegisterToCurrentSeasonJob` to help set up test data, so introducing the notion that there are periods of time where the app is actually out of any season may end up breaking any test using these factories. 

The code also liberally uses `Season.current`, which doesn't work any more if there are times where there's no season. I can try returning some sort of null season object after the season end, but I think it's going to break a lot of things.

I'll keep thinking about this but at the moment it's looking like always being in some season is foundational to the app.
